specification/app/resource-manager/Microsoft.App/preview/2022-11-01-preview/ContainerAppsJobs.json

```diff
@@ -0,0 +1,781 @@
+                "create",
+                "read",
+                "update"
+              ]
+            },
+            "configuration": {
+              "$ref": "#/definitions/JobConfiguration",
+              "description": "Container Apps Job configuration properties."
+            },
+            "template": {
+              "$ref": "./CommonDefinitions.json#/definitions/ContainerAppsJobTemplate",
+              "description": "Container Apps job definition."
+            },
+            "outboundIpAddresses": {
+              "description": "Outbound IP Addresses of a container apps job.",
+              "type": "array",
+              "items": {
+                "type": "string"
+              }
+            },
+            "eventStreamEndpoint": {
+              "description": "The endpoint of the eventstream of the container apps job.",
+              "type": "string"
+            }
+          }
+        }
+      }
+    },
+    "ContainerAppsJobsCollection": {
+      "description": "Container Apps Jobs collection ARM resource.",
+      "required": [
+        "value"
+      ],
+      "type": "object",
+      "properties": {
+        "value": {
+          "description": "Collection of resources.",
+          "type": "array",
+          "items": {
+            "$ref": "#/definitions/ContainerAppsJob"
+          }
+        },
+        "nextLink": {
+          "description": "Link to next page of resources.",
+          "type": "string",
+          "readOnly": true
+        }
+      }
+    },
+    "RegistryCredentials": {
+      "description": "Container App Private Registry",
+      "type": "object",
+      "properties": {
+        "server": {
+          "description": "Container Registry Server",
+          "type": "string"
+        },
+        "username": {
+          "description": "Container Registry Username",
+          "type": "string"
+        },
+        "passwordSecretRef": {
+          "description": "The name of the Secret that contains the registry login password",
+          "type": "string"
+        },
+        "identity": {
+          "description": "A Managed Identity to use to authenticate with Azure Container Registry. For user-assigned identities, use the full user-assigned identity Resource ID. For system-assigned identities, use 'system'",
+          "type": "string"
+        }
+      }
+    },
+    "JobSecretsCollection": {
+      "description": "Container Apps Job Secrets Collection ARM resource.",
+      "required": [
+        "value"
+      ],
+      "type": "object",
+      "properties": {
+        "value": {
+          "description": "Collection of resources.",
+          "type": "array",
+          "items": {
+            "$ref": "./CommonDefinitions.json#/definitions/Secret"
+          },
+          "x-ms-identifiers": [
+            "name"
+          ]
+        }
+      }
+    },
+    "ReplicaCount": {
+      "description": "Minimum number of successful replica completions before overall job completion.",
```

Comment:
This description is different than spec. In spec this is the number of replicas in total to run in a job execution.

Reply:
@anthonychu we discussed on this and updated this as per the following suggestion: https://github.com/Azure/azure-rest-api-specs/pull/21846/files/feaac967641b198543cd57b4fb2644b3b3a80a18#diff-739c3ca30297197a20a321984788ec7f5189f3f637da962aae461dc834f5613c

@trajkobal do we need to revert it back?

Reply:
If this maps to `completions` in k8s, then I think `replicaCount` is not the right name for it. I suggest we consider `completions` or `replicaCompletionCount`.

Reply:
I agree with you @anthonychu. Now we have to convince @vinisoto and @anandanthony

Reply:
> @anthonychu we discussed on this and updated this as per the following suggestion: https://github.com/Azure/azure-rest-api-specs/pull/21846/files/feaac967641b198543cd57b4fb2644b3b3a80a18#r1048686943
> 
> @trajkobal do we need to revert the description back?

No, unless we have a better way to explain it. The spec description seemed a bit obscure to me.

Reply:
@anthonychu we discussed on the description given here. Basically, the property "replicaCount" would be used to define the minimum number of replicas that a job would get created for a job and completed on a job's execution. In case anyone of the replicas fail, another replica would get created and executed. This would happen till all the replicas (as per the replicaCount) have completed their execution (before replicaTimeout). Hence, we went ahead with this description and name.

Reply:
@anthonychu updated this property as per our discussion.